### PR TITLE
Fix the page token bug for all the entities.

### LINKED ISSUE #372: [CLI] Page token not working when listing the entities
**Describe the bug**

"Invalid parameters" message shows up when `page_token` option is passed in CLI while listing the entities.

**To Reproduce**

1. `bin/uc catalog create --name test`
2. `bin/uc catalog list --max_results 1 --page_token test`

**Expected behavior**

Passing this option should output a proper list of the entities - as per the next page token provided - instead of the helper message.

**System [please complete the following information]:**

-   OS: macOS Sonoma 14.5

**Additional context**

This has to be resolved for all the entities.

### PR DESCRIPTION
**Pull request by vksx**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

- Fixed the page token bug for all the entities - Catalog, Schema, Table, Volume, Function and Model. Now passing the page token will list the the entities properly.

  Example:
  ```
  bin/uc catalog create --name test
  bin/uc catalog list --page_token test
  ```

- `listVolumes()` in the VolumeRepository now utilizes PagedListingHelper to list and paginate the volumes.

Fixes #372

### CODE REVIEW

**Review by vikrantpuppala — APPROVED** [review]
thank you vm for this! @rameshchandra to merge

**Comment by rameshchandra** [comment]
@vksx can you add a pagination test?

**Comment by vksx** [comment]
Thanks for reviewing this PR!
@rameshchandra Added the pagination tests for all the entities. PTAL.
@vikrantpuppala I also added the page token fix for the model version which was not included in the first commit.

**Review by vikrantpuppala — APPROVED** [review]
@rameshchandra can you take a look please

File: server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java [hunk 1]
```diff
@@ -47,24 +47,38 @@ public void testCatalogCRUD() throws ApiException {
...
     System.out.println("Testing list catalogs..");
-    List<CatalogInfo> catalogList = catalogOperations.listCatalogs();
+    List<CatalogInfo> catalogList = catalogOperations.listCatalogs(Optional.empty());
     assertThat(catalogList).isNotNull();
     assertCatalogExists(catalogList, CATALOG_NAME, COMMENT, PROPERTIES);
+    assertCatalogExists(catalogList, COMMON_ENTITY_NAME, COMMENT, PROPERTIES);
+
+    // List catalogs with page token
+    System.out.println("Testing list catalogs with page token..");
+    catalogList = catalogOperations.listCatalogs(Optional.of(CATALOG_NAME));
```
Comment (rameshchandra) [inline server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java:65]: Does this test implicitly assume that the page token is the catalog name? The page token is intended to be opaque with the client not interpreting it. Can you rewrite the test to do the previous `listCatalogs` call with max result of 1 and use the returned page token to do another `listCatalogs` call?

File: server/src/test/java/io/unitycatalog/server/base/function/BaseFunctionCRUDTest.java [hunk 2]
```diff
@@ -101,7 +141,29 @@ public void testFunctionCRUD() throws ApiException {
                   .anySatisfy(parameter -> assertThat(parameter.getName()).isEqualTo("param1"));
             });
 
+    // List functions with page token
+    System.out.println("Testing list functions with page token..");
+    functionInfos =
+        functionOperations.listFunctions(CATALOG_NAME, SCHEMA_NAME, Optional.of(FUNCTION_NAME));
```
Comment (rameshchandra) [inline server/src/test/java/io/unitycatalog/server/base/function/BaseFunctionCRUDTest.java:147]: Same comment as for `listCatalogs` -- can you do the test without assuming that the page token is the function name?

**Review by rameshchandra — APPROVED** [review]
LGTM. 

Thank you @vksx for fixing the issue. Can you please update the tests to not assume a specific format for the page tokens as pointed out in the comments?

File: server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java (hunk 1 above)
Comment (vksx) [reply server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java:65]: The page token being used is the [name of the last entity](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/server/src/main/java/io/unitycatalog/server/persist/utils/PagedListingHelper.java#L61) of the page and is contained in the [properties of `ListCatalogsResponse`](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/api/Models/ListCatalogsResponse.md) (and equivalent responses for other entities). But, the [tests are using `List<CatalogInfo>`](https://github.com/unitycatalog/unitycatalog/blob/141936063f3aae6aeb0185110b0731a7059ec64a/server/src/test/java/io/unitycatalog/server/base/catalog/CatalogOperations.java#L13) which, as expected, does not contain any information regarding the `next_page_token` - so for brevity (as the PR is already big) I used the entity name directly. 

I can make the changes and replace `List<CatalogInfo>` with `ListCatalogsResponse` (and do the same for all other entities) in this same PR or if necessary, I can create a new PR for these changes.

Please let me know if I am missing something or if we should continue with the current approach.

Thanks!

**Comment by vksx** [comment]
In case someone is seeing the mess I have created here - I am really sorry - I recently found out in the latest merge that I need to sign off my previous commits to this PR (DCO) - so I followed the instructions mentioned in the DCO details page and this happened after I resolved the "merge conflicts" that came up during the rebase and force committed. 

I am trying to fix this as soon as I can. I sincerely apologise for this.

File: server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java (hunk 1 above)
Comment (rameshchandra) [reply server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java:65]: Thanks! It is ok to do it in a new PR.

**Review by rameshchandra — APPROVED** [review]

**Comment by vikrantpuppala** [comment]
@vksx can you sign off your commits on this one and we can merge this. It does get messy when running the commands given in DCO, but it is easy if you squash your commits first. How I generally do this is:
```
git merge-base HEAD <branch-from-where-you-forked-this-branch>
git reset --soft <commit-hash-from-above-command>
git commit -am "commit message"
git rebase HEAD~1 --signoff
git push --force
```

**Pull request merged**